File: Second Seminar/SecondSeminar/src/main/java/com/example/SecondSeminar/post/exception/PostExceptionType.java
package com.example.SecondSeminar.post.exception;

import com.example.SecondSeminar.common.exception.ExceptionType;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;

@RequiredArgsConstructor
public enum PostExceptionType implements ExceptionType {
    NOT_FOUND_POST(HttpStatus.NOT_FOUND, "존재하지 않는 게시글입니다.");

    private final HttpStatus httpStatus;
    private final String message;

    @Override
    public HttpStatus httpStatus() {
        return httpStatus;
    }

    @Override
    public String message() {
        return message;
    }
}

File: Second Seminar/SecondSeminarKotlin/src/main/kotlin/com/example/SecondSeminarKotlin/post/exception/PostExceptionType.kt
package com.example.SecondSeminarKotlin.post.exception

import com.example.SecondSeminarKotlin.exception.ExceptionType
import org.springframework.http.HttpStatus
import org.springframework.http.HttpStatus.*

enum class PostExceptionType(
    override val httpStatus: HttpStatus,
    override val message: String
) : ExceptionType {
    NOT_FOUND_POST(NOT_FOUND, "존재하지 않는 게시글입니다")
}